cohere/responses/generation.py

```diff
@@ -114,3 +115,22 @@ def prompt(self) -> str:
         """Returns the prompt used as input"""
         return self[0].prompt  # should all be the same
 
+
+StreamingTokenLikelihood = NamedTuple("TokenLikelihood", [("index", int), ("token", str), ("likelihood", float)])
+
+
+class StreamingGenerations(CohereObject):
+
+    def __init__(self, response):
+        self.response = response
+
+    def __iter__(self) -> Generator[StreamingTokenLikelihood, None, None]:
+        if not isinstance(self.response, requests.Response):
+            raise ValueError("For AsyncClient, use `async for` to iterate through the `StreamingGenerations`")
```

Comment:
do we need symmetrical check in async version?

Reply:
I think this mistake is far more likely